pypdf/_writer.py

```diff
@@ -2095,7 +2095,7 @@ def addLink(
         deprecate_with_replacement(
             "addLink", "add_annotation(AnnotationBuilder.link(...))", "4.0.0"
         )
-        return self.add_link(pagenum, page_destination, rect, border, fit, *args)
+        self.add_link(pagenum, page_destination, rect, border, fit, *args)
```

Comment:
```suggestion
        return self.add_link(pagenum, page_destination, rect, border, fit, *args)
```

Reply:
Why did you remove the return?

(Not that it really matters due to the deprecation message, but I would prefer to not make changes to deprecated methods)

Reply:
if you want so we need to change the annotation : as it is deprecated I preferred to not change it.
Your choice

Reply:
Ah, got it! Sorry, I missed that part :+1:  Thanks for not blindly following my suggestion :pray: